modules/Movecraft/src/main/java/net/countercraft/movecraft/async/translation/TranslationTask.java

```diff
@@ -77,26 +82,77 @@ protected void execute() {
         if (dz != preTranslateEvent.getDz()) {
             dz = preTranslateEvent.getDz();
         }
+        if (world != preTranslateEvent.getWorld()) {
+        	world = preTranslateEvent.getWorld();
+        }
         final int minY = oldHitBox.getMinY();
         final int maxY = oldHitBox.getMaxY();
 
+        // proccess nether portals
+        if (Settings.CraftsUseNetherPortals && craft.getW().getEnvironment() != Environment.THE_END && world.equals(craft.getW())) {
+        	
+        	// ensure chunks are loaded for portal checking only if change in location is large
+        	List<List<Object>> chunksToLoad = craft.checkChunks(craft.getChunks(2, 2, dx, dy, dz, world));
+            if (!chunksToLoad.isEmpty()) {
+            	
+            	synchronized (this) {
```

Comment:
I'm not sure if I understand the intent of this. If you're trying to load chunks async, wouldn't it be better to handle this on the implementation side rather than the calling side?

Reply:
the synchronized block here is necissary in order to use the `this.wait()` that forces the task to wait for the chunks to be loaded before continuing. It is kind of misleading and has nothing to do with being synchronized with the main server thread.

Reply:
I believe it is explained in the java docs for Object#notify()

Reply:
yes but by default chunk loading in spigot is not async and blocks the executing thread regardless, unless you're specifically doing this for async chunkloading support in paper?

Reply:
The ChunkManager#loadChunks() method creates a task that is run on the next server tick, which does not block the thread, so I had to block the thread manually until that task is complete. Basically if I don't it might try to keep going before the next server tick happens and the chunks are loaded which is what the problem is.

Reply:
Instead of using notify in `ChunkManager#loadChunks()`, could you use `BukkitScheduler#callSyncMethod` and wait (`get`) on the future? Than you should be able to avoid synchronized blocks. A concern would be ensuring that users don't call it on the main thread.

Reply:
I was not aware of that method, I can see the usefulness of it. Change has been made.